service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/NarrativeStatementMapper.java

```diff
@@ -19,19 +19,27 @@
 @RequiredArgsConstructor(onConstructor = @__(@Autowired))
 public class NarrativeStatementMapper {
 
-    private final MessageContext messageContext;
-
     private static final Mustache NARRATIVE_STATEMENT_TEMPLATE = TemplateUtils.loadTemplate("ehr_narrative_statement_template.mustache");
 
+    private final MessageContext messageContext;
+    private final ParticipantMapper participantMapper;
+
     public String mapObservationToNarrativeStatement(Observation observation, boolean isNested) {
+        final IdMapper idMapper = messageContext.getIdMapper();
         var narrativeStatementTemplateParameters = NarrativeStatementTemplateParameters.builder()
-            .narrativeStatementId(messageContext.getIdMapper().getOrNew(ResourceType.Observation, observation.getId()))
+            .narrativeStatementId(idMapper.getOrNew(ResourceType.Observation, observation.getId()))
             .availabilityTime(getAvailabilityTime(observation))
             .comment(observation.getComment())
-            .isNested(isNested)
-            .build();
+            .isNested(isNested);
+
+        if (observation.hasPerformer()) {
+            final String participantReference = idMapper.get(observation.getPerformerFirstRep());
```

Comment:
we could filter out the Practitioner reference only from the performers list.

Reply:
Sorry, I don't understand what you mean here - could you elaborate please @msolecka?

Reply:
sorry double checked docs and analysed it and noticed I was wrong.